desdeo_emo/selection/APD_Select_constraints.py

```diff
@@ -83,6 +83,28 @@ def do(self, pop: Population, vectors: ReferenceVectors) -> List[int]:
             sub_population_index = np.atleast_1d(
                 np.squeeze(np.where(assigned_vectors == i))
             )
+
+            # Constraint check
+            if len(sub_population_index) > 1 and pop.constraint_violation is not None:
+                constraint_following_index = []
+                violation_values = []
+                for individual in sub_population_index:
+                    violation_values.append(pop.constraint_violation[individual])
+
+                    if((pop.constraint_violation[individual] > 0).all()):
+                        constraint_following_index.append(individual)
+
+                violation_values = np.asarray(violation_values)
+                violation_values = violation_values.sum(0)
+
+                # Case when entire subpopulation is infeasible
+                if len(constraint_following_index) == 0:
+                    sub_population_index = sub_population_index[np.where(violation_values == violation_values.max())]
```

Comment:
Another reason why making feasible solutions have negative "violation_values" would be better: it looks like you're maximizing the infeasibility here, if you just read the line without the context above.

So, just make the assumption that negative violations are better, and do violation_values.min()